azure-devops/azext_devops/dev/common/_credentials.py

```diff
@@ -29,18 +32,46 @@ def set_credential(devops_organization, token):
     cred_store = CredentialStore()
     cred_store.set_password(key, token)
 
+    if os.path.isfile(_ORGANIZATION_LIST_FILE):
+        # No need to add organization if it's already present.
+        with open(_ORGANIZATION_LIST_FILE, 'r') as org_list:
+            for organization in org_list:
+                if key == organization.rstrip():
+                    return
+    with open(_ORGANIZATION_LIST_FILE, 'a+') as org_list:
+        org_list.write(key + "\n")
+
 
 def clear_credential(devops_organization):
     key = _get_service_name(devops_organization)
     logger.debug('Clearing credential: %s', key)
     cred_store = CredentialStore()
-    cred_store.clear_password(key)
+    
+    if key == _DEFAULT_CREDENTIAL_KEY:
+        # remove all organizations and delete the file
+        if os.path.isfile(_ORGANIZATION_LIST_FILE):
+            with open(_ORGANIZATION_LIST_FILE) as org_list_file:
+                for organization in org_list_file:
+                    cred_store.clear_password(organization.rstrip())
+            os.remove(_ORGANIZATION_LIST_FILE)
+        else:
+            raise CLIError('No credentials were found.')
```

Comment:
Breaking back compat: 
User will have stored creds but no file, after the upgrade user runs logout command. He will get no credentials were found and creds will not be cleared.

Reply:
I agree, but what's the alternative here? We can't know what credentials were stored by user before the upgrade. The reason being keyring or credential manager doesn't provide a list of keys/credentials.

Reply:
Agree we cannot delete all keys. 

Lets try to clear with default key atleast if it is present we will clean it.. if not, the message no credentials were found is also valid as per earlier behavior. 
We will end up covering what was supported earlier i.e. clearing the default key.

Reply:
Yea, makes sense to at least clear the old default credential. Modified 'clear_credential' function. Thanks